src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/XmlFormatReaderGenerator.cs

```diff
@@ -98,17 +98,19 @@ private class CriticalHelper
             private ArgBuilder _memberNamesArg;
             private ArgBuilder _memberNamespacesArg;
             private ArgBuilder _collectionContractArg;
-
-            private bool _useReflection = true;
+#endif
 
             public XmlFormatClassReaderDelegate GenerateClassReader(ClassDataContract classContract)
             {
-                if (_useReflection)
+                if (DataContractSerializer.Option == SerializationOption.ReflectionAsBackup || DataContractSerializer.Option == SerializationOption.ReflectionOnly)
                 {
```

Comment:
Is the above condition correct? Do we want to execute the 'else' block only when DataContractSerializer.Option == SerializationOption.CodeGenOnly?